### ISSUE TITLE
bug: cannot exit replace mode in Jupyter notebook

### ISSUE THREAD

**Issue by penelopeysm** [vscode-neovim/vscode-neovim#1304]
### Did you check docs and existing issues?

- [X] I have read all the vscode-neovim docs
- [X] I have searched the existing issues of vscode-neovim
- [X] I can reproduce the issue with an empty init.vim/init.lua and no other vscode extensions (when applicable)

### Neovim version (nvim -v)

0.9.1

### Operating system/version

macOS 13.4.1

### Describe the bug

Entering replace mode in a Jupyter notebook cell and then pressing `<Esc>` doesn't go back to normal mode.

Following this, it's impossible to return to normal mode when re-entering the cell, until VSCode is restarted.

This bug doesn't happen in normal text files, only notebooks as far as I can tell.

Disabling VSCode's <Esc> keyboard shortcut for exiting the cell doesn't fix the bug (the cell is no longer exited, but it's still impossible to return to normal mode).

--------

Quick video (all VSCode extensions disabled and nvim config removed):

https://github.com/vscode-neovim/vscode-neovim/assets/122629585/d3f8a2ef-d85f-4ab8-a1d7-a418ed2ac6d7

### Steps To Reproduce

1. Create a new Jupyter notebook in VSCode.
2. Start editing a cell.
3. Press `R` to enter replace mode
4. Press `Esc`

### Expected Behavior

Return to normal mode.

**Comment by sleepycat233** [comment]
I'm still encountering issues with exiting the single character replacement mode. When I press `r` and then `Esc`, it triggers the `notebook.cell.quitEdit` command and exits the cell, instead of just exiting the single character replacement mode. I've observed a similar behavior when using `leap.neovim`. If I press `Esc` after `s`, it exits the cell editing mode, but the decorators remain visible on the screen.

As a temporary solution, I've removed the `!notebookCellFocused` condition from `vscode-neovim.escape` and reassigned the `notebook.cell.quitEdit` command to a different keybind, such as `shift+escape`.
